Tests/Grid/Mapping/Metadata/MetadataTest.php

```diff
@@ -0,0 +1,236 @@
+    }
+
+    public function testSetterFieldMappingWithField()
+    {
+        $fieldRandom = 'foo';
+        $valueRandom = 'bar';
+        $fieldMapping = [$fieldRandom => $valueRandom];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+
+        $this->assertAttributeEquals(
+            $fieldMapping,
+            'fieldsMappings',
+            $this->metadata
+        );
+    }
+
+    public function testGetterFieldMappingWithField()
+    {
+        $fieldRandom = 'foo';
+        $valueRandom = 'bar';
+        $fieldMapping = [$fieldRandom => $valueRandom];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+
+        $this->assertTrue($this->metadata->hasFieldMapping($fieldRandom));
+    }
+
+    public function testGetterFieldMappingReturnDefaultTypeText()
+    {
+        $fieldRandom = 'foo';
+        $valueRandom = 'bar';
+        $fieldMapping = [$fieldRandom => $valueRandom];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+
+        $this->assertEquals(
+            'text',
+            $this->metadata->getFieldMappingType($fieldRandom)
+        );
+    }
+
+    public function testSetterMappingFieldWithType()
+    {
+        $fieldRandom = 'foo';
+        $valueRandom = 'bar';
+        $fieldMapping = [
+            $fieldRandom => [
+                'type' => $valueRandom
+            ]
+        ];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+
+        $this->assertAttributeEquals(
+            $fieldMapping,
+            'fieldsMappings',
+            $this->metadata
+        );
+    }
+
+    public function testGetterMappingFieldWithType()
+    {
+        $field = 'foo';
+        $value = 'bar';
+        $fieldMapping = [
+            $field => [
+                'type' => $value
+            ]
+        ];
+
+        $this->metadata->setFieldsMappings($fieldMapping);
+        $this->assertEquals($value, $this->metadata
+            ->getFieldMappingType($field));
+    }
+
+    public function testSetterGroupByWithString()
+    {
+        $groupBy = 'groupBy';
+
+        $this->metadata->setGroupBy($groupBy);
+
+        $this->assertAttributeEquals(
+            $groupBy,
+            'groupBy',
+            $this->metadata
+        );
+    }
+
+    public function testGetterGroupByWithString()
+    {
+        $groupBy = 'groupBy';
+
+        $this->metadata->setGroupBy($groupBy);
+        $this->assertEquals($groupBy, $this->metadata->getGroupBy());
+    }
+
+    public function testSetterNameWithString()
+    {
+        $name = 'name';
+
+        $this->metadata->setName($name);
+
+        $this->assertAttributeEquals(
+            $name,
+            'name',
+            $this->metadata
+        );
+    }
+
+    public function testGetterNameWithString()
+    {
+        $name = 'name';
+
+        $this->metadata->setName($name);
+
+        $this->assertEquals($name, $this->metadata->getName());
+    }
+
+    public function testSetterColumnsFromMapping()
```

Comment:
If you're testing `getter` behavior, `setterColumnsFromMapping` is not a good one. I can't check code right now but I'm sure you can find a better name ;)